File: reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        MIN = -2**31
        MAX = 2**31 - 1
        result = 0
        is_negative = x < 0

        x = abs(x)

        while x != 0:
            digit = x % 10
            x = x // 10
            
            if result > (MAX - digit) // 10:
                return 0

            result = result * 10 + digit

        return -result if is_negative else result

File: reverse-integer/reverse-integer.ts
function reverse(x: number): number {
    let y = '';
    const MIN = -2147483648;
    const MAX = 2147483647;
    let isNegative = x < 0;
    
    while (x !== 0) {
        y += Math.abs(x % 10);
        x = Math.trunc(x / 10);
    }

    let result = Number(y);
    if (isNegative) result = result * -1;

    if (result < MIN || result > MAX) {
        return 0;
    }

    return result;
}